src/Orckestra.Composer.Cart/Factory/Order/OrderDetailsViewModelFactory.cs

```diff
@@ -138,32 +138,30 @@ protected virtual OrderDetailInfoViewModel GetOrderInfosViewModel(CreateOrderDet
             orderInfos.OrderStatusRaw = param.Order.OrderStatus;
             orderInfos.BillingCurrency = param.Order.Cart.BillingCurrency;
             orderInfos.PricePaid = LocalizationProvider.FormatPrice((decimal)param.Order.Cart.Total, CurrencyProvider.GetCurrency());
-            orderInfos.IsOrderEditable = IsOrderEditable(param);
+            orderInfos.HasOwnDraft = HasOwnDraft(param);
 
             return orderInfos;
         }
 
-        protected virtual bool IsOrderEditable(CreateOrderDetailViewModelParam param)
+        protected virtual bool HasOwnDraft(CreateOrderDetailViewModelParam param)
         {
-            var shipmentStatuses = param.Order.Cart.GetAllShipmentStatuses();
-            if (!shipmentStatuses.Any() 
-                || param.OrderSettings == null
-                || string.IsNullOrWhiteSpace(param.OrderSettings.EditableShipmentStates))
+            //TODO - For now there is no nice way to identify if website user owns draft or not
+            // For now we use possible owner names we invetigated, which can be used in AuthToken
+            // need to wait platform fix, to have nice way to identify if user own the dtaft
+            var orderDraft = param.OrderCartDrafts?.FirstOrDefault(d => Guid.Parse(d.Name) == Guid.Parse(param.Order.Id));
```

Comment:
Is it good that it will have a format exception if the cart name is not Guid?(like"Default")

Looks like  Guid.Parse(param.Order.Id) called too many times...

Reply:
fixed